lib/streams.js

```diff
@@ -54,3 +54,78 @@ class WriteStream extends Writable {
 }
 
 exports.WriteStream = WriteStream
+
+class ByteStream extends Readable {
+  constructor (core, opts = {}) {
+    super()
+
+    this._core = core
+    this._index = 0
+    this._range = null
+
+    this._byteOffset = opts.byteOffset || 0
+    this._byteLength = typeof opts.byteLength === 'number' ? opts.byteLength : -1
+    this.prefetch = typeof opts.prefetch === 'number' ? opts.prefetch : 32
+
+    this._applyOffset = this._byteOffset > 0
+  }
+
+  _open (cb) {
+    this._openp().then(cb, cb)
+  }
+
+  _read (cb) {
+    this._readp().then(cb, cb)
+  }
+
+  async _openp () {
+    if (this._byteLength === -1) {
+      await this._core.update()
+      this._byteLength = Math.max(this._core.byteLength - this._byteOffset, 0)
+    }
+  }
+
+  async _readp () {
+    let data = null
+
+    if (this._byteLength === 0) {
+      this.push(null)
+      return
+    }
+
+    let relativeOffset = 0
+
+    if (this._applyOffset) {
+      this._applyOffset = false
+
+      const [block, byteOffset] = await this._core.seek(this._byteOffset)
+
+      this._index = block
+      relativeOffset = byteOffset
+    }
+
+    this._prefetch(this._index + 1)
+    data = await this._core.get(this._index++)
+
+    if (relativeOffset > 0) data = data.subarray(relativeOffset)
+
+    const final = data.length >= this._byteLength
+    if (final) data = data.subarray(0, this._byteLength)
+    this._byteLength -= data.length
```

Comment:
if (data.byteLength > this._byteLength) data = data.subarray(0, this._byteLength)
this._byteLength -= data.byteLength

...


if (this._byteLength === 0) this.push(0)

Reply:
Can't believe that is equivalent lol